src/mono/mono/mini/aot-compiler.c

```diff
@@ -5252,8 +5255,14 @@ MONO_RESTORE_WARNING
 				mono_error_assert_ok (error);
 
 				add_method (acfg, wrapper);
-				if (export_name)
+				if (export_name) {
 					g_hash_table_insert (acfg->export_names, wrapper, export_name);
+#ifdef TARGET_APPLE_MOBILE
+					g_string_append_printf (export_symbols, "_%s\n", export_name);
+#else
+					g_string_append_printf (export_symbols, "%s\n", export_name);
```

Comment:
For what other platforms should we anticipate using this, Android, Wasm, any others?

Reply:
This needs a separator between the symbols.

Reply:
its writing each symbol on a new line in the output file, so symbols are separated that way, similar to how NativeAOT produce platform specific exports files in https://github.com/dotnet/runtime/blob/b2dc37ba181a7fa4427e717eab819ba3543d0ae4/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ExportsFileWriter.cs#L13.

Reply:
Alternative is to use a list instead holding the symbols and push the formatting when writing the file, that would be a more flexible approach.

Reply:
> For what other platforms should we anticipate using this, Android, Wasm, any others?

Regarding Mitchell's comment I have opened an issue: https://github.com/dotnet/runtime/issues/79491